Makefile

```diff
@@ -286,8 +287,45 @@ compile_clarity:
 	fi
 	@echo "Done."
 
+compile_clarity_harborui:
+	@echo "compiling binary for clarity ui with local harbor ui library..."
+	@if [ ! -e $(UINGPATH)/package.json.org ] ; then \
+		cp $(UINGPATH)/package.json $(UINGPATH)/package.json.org; \
+		sed -i -e "s/\"harbor-ui\": \"\^[0-9].\+\"/\"harbor-ui\": \"file:.\/lib\/\"/g" $(UINGPATH)/package.json; \
+	fi
+	@if [ "$(HTTPPROXY)" != "" ] ; then \
+		$(SEDCMD) -i 's/__proxy__/--proxy $(HTTPPROXY)/g' $(DOCKERFILE_BUILD_UI) ; \
+		$(DOCKERBUILD) -f $(DOCKERFILE_BUILD_UI) -t $(DOCKERIMAGENAME_CLATIRY):dev .; \
```

Comment:
please don't hard code the image tag here, you can use commit id like https://github.com/vmware/harbor/blob/master/make/photon/Makefile#L55 does.